### ISSUE TITLE
Input Description Text

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1641]
## Overview

`TextField` should accept a new optional `description` property that is displayed immediately below the label text. When `description` is present, the associated input field should have `aria-describedby` with an idref pointing to the associated description.

## Related Components

- https://github.com/dequelabs/cauldron/issues/1656
- https://github.com/dequelabs/cauldron/issues/1658
- https://github.com/dequelabs/cauldron/issues/1657

### Visual Design

Figma: https://www.figma.com/design/CEFVdiecqDjLSjhorjHUzI/Product-Foundations-(Cauldron)-Library?node-id=50-1992&t=UIec9KMPGKQia7fz-4

![textfield design](https://github.com/user-attachments/assets/62405b8c-c667-4ab8-aa8a-e4b93fdaf6c7)
![textfield multiline design](https://github.com/user-attachments/assets/3ce2f484-9511-4fc5-b4c0-64e820cf92ae)

When the "label text" overflows with the "required" label, the label should wrap onto a new line without interfering with the placement or sizing of required.

> [!NOTE]
> This should match the description text that is current present on `RadioGroup`

### Interface / Props

```tsx
interface TextFieldProps {
  // ...other props
  description?: React.ReactNode
}
```

### Accessibility

> [!NOTE]
> When both `description` and `error` are set, the `aria-describedby` property should reference the description idref first, followed by the error idref.

## Additional Changes

With this change, the placement of the error message should also be visually located immediately above the input field with the addition of a `caution` icon preceding the error text.

The _Required_ text should not have any font styles applied and have italics removed.

**Comment by shawnsharpDQ** [comment]
![Description and Error Updates](https://github.com/user-attachments/assets/3d918083-9c2e-468c-a316-ceb8a8c40582)

**Comment by scurker** [comment]
Each component should be tackled separately, with a `FieldGroup` component that accepts a label, description, and error state.

**Comment by amberfrontend** [comment]
I have a fix for this, but I'm not sure I have permission to push up a PR.

**Comment by scurker** [comment]
Hey @amberfrontend 👋 

I believe you should have permissions to create a PR, but it might need to be from a fork. If that doesn't work let me know and I can double check the contributor settings.

**Comment by scurker** [comment]
@amberfrontend I'd like to help but I'm not sure I know how to help. We've had outside contributors contribute from a fork before (such as this one: https://github.com/dequelabs/cauldron/pull/1464). I checked our contributor settings and it should still allow for outside contributors. Are there any errors or issues that you are seeing that you can share?

**Comment by scurker** [comment]
@ezirlingerDQ What's the expected scenario when the label text needs to wrap?

**Comment by Amandeque** [comment]
Team Drums Estimation: 3 Drum Points

**Comment by DanylAksonov** [comment]
> [@DanylAksonov](https://github.com/DanylAksonov) , could you please provide test notes for this ticket.

@Keerthi-Penukonda, I think you can check the TextField component [documentation](https://cauldron.dequelabs.com/components/TextField) and verify the following points:
• The description displays correctly according to the ticket requirements (color, size, position).
• The 'Required' label for the select field is not italic.
• The error message for the TextField displays correctly (color, size, position).

**Comment by Keerthi-Penukonda** [comment]
### TextField should accept a new optional description property that is displayed immediately below the label text. When description is present, the associated input field should have aria-describedby with an idref pointing to the associated description.

<details><summary>Working fine</summary>

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e7736073-b800-4dfb-aff3-95963a0ee1fb" />

</details>

### When the "label text" overflows with the "required" label, the label should wrap onto a new line without interfering with the placement or sizing of required.

<details><summary>Working fine - However required text is not aligned properly</summary>

<img width="1058" height="346" alt="Image" src="https://github.com/user-attachments/assets/29188e77-44c2-4512-8d4b-df12127ed833" />

</details>

### This should match the description text that is current present on RadioGroup

<details><summary>Text is different - I'm unsure about this</summary>

<img width="1061" height="409" alt="Image" src="https://github.com/user-attachments/assets/0cbd3fd7-84da-4b52-8f54-6d4aba3b9a4e" />

<img width="1048" height="412" alt="Image" src="https://github.com/user-attachments/assets/c363f8e6-fffe-4b7d-8c56-0cf44b6eb889" />

</details>

### When both description and error are set, the aria-describedby property should reference the description idref first, followed by the error idref.

There is no such field to validate(With both description and error)

### With this change, the placement of the error message should also be visually located immediately above the input field with the addition of a caution icon preceding the error text.

There is no such field to validate(With both description and error)

### The Required text should not have any font styles applied and have italics removed.

There is no such field to validate(With both description and required)

### Multi-line with description

We don't have such field to validate
